In [1]:
import pandas as pd
import os
from os import path

In [2]:
filePath = path.join(os.getcwd(), "ChatGPT_Tweets_Dataset_Full.csv")

In [3]:
df = pd.read_csv(filePath)

In [4]:
df.shape

(3821843, 17)

### check which columns are interesting and how

get just the first 1000 rows to work with a more managable dataset

In [5]:
dfTemp = df.drop(columns=['Unnamed: 0', "original_text"]).head(1000)

dfTemp.to_csv("1000.csv", index=False)

In [6]:
dfTemp["tasks"].value_counts()

tasks
have a moving chat              12
optimizing language models      11
generated an entire story        9
have a chat                      3
here refer to the screenshot     2
                                ..
prompting results                1
explain a concept                1
personalize to the user          1
packaged prompts                 1
explaining concepts              1
Name: count, Length: 117, dtype: int64

In [7]:
dfTemp["technologies"].value_counts() #INTERESTING

technologies
model                   84
api                     47
chatbot                 16
google                  12
internet                12
                        ..
deep learning, model     1
c                        1
platforms, model         1
aiml                     1
platforms                1
Name: count, Length: 88, dtype: int64

In [8]:
dfTemp["organizations"].value_counts() 

organizations
GPT3                                                                                               238
GPT                                                                                                 50
AI, GPT3 Heres                                                                                      37
gpt                                                                                                 23
gpt3                                                                                                18
                                                                                                  ... 
Gpt3, Naruto                                                                                         1
GPT3 Final                                                                                           1
Fermat, Use GPT3 Stable Difussion, Lexica                                                            1
GPT3, AI MachineLearning DataScience, ArtificialIntelligenc

In [9]:
dfTemp["job_competencies"].value_counts() #INTERESTING

job_competencies
plan                                                                           36
terminology                                                                    32
use personal organization software                                             21
communication                                                                  16
work independently                                                             14
                                                                               ..
patents, use personal organization software                                     1
acoustics, computer programming, work independently                             1
patents, LESS, Python (computer programming)                                    1
electrical machines, ML (computer programming), review unpublished articles     1
English, statistics, terminology, LESS                                          1
Name: count, Length: 190, dtype: int64

In [10]:
dfTemp["job_profiles"].value_counts() #INTERESTING

job_profiles
secretary                                 42
drawing artist                            24
prompter                                  23
writer                                    15
armed forces officer                      10
building cleaner                           8
product assembly inspector                 4
tourist guide                              3
store detective                            3
blogger                                    2
video and motion picture editor            2
statistical assistant                      2
medium                                     2
performance artist                         2
auctioneer                                 2
prompter, armed forces officer             1
stand-in                                   1
ICT trainer                                1
companion                                  1
government minister                        1
economist                                  1
chocolatier                               

In [11]:
dfTemp["nouns"].value_counts() 

nouns
today, ai, w, gpt3, plan, playground, connect, ship, web, app, chrome, extension, username    35
openai, upgrades, poetry, lyrics                                                              21
openai, upgrades, poetry, andlyrics                                                            9
v1, fork, gpt6b, tokens                                                                        8
chatgpt, language, models, dialogue                                                            6
                                                                                              ..
fire, copywriters, year, gpt3                                                                  1
bunch, apps, gpt3, prompts, buttons, heels, fund                                               1
gpt3, transformer, language, model, text, im, iteration, content                               1
office, email, cool                                                                            1
chatgpt, thing, educatio

In [12]:
dfTemp["verbs"].value_counts()

verbs
im, working, building, writing, heres, explore, gpt3s, seems, wan, build, check    35
gpt3, stunning, rhyming                                                            29
is                                                                                 22
releasing, finetuned                                                               11
optimizing                                                                         10
                                                                                   ..
npews, written, know                                                                1
argues, are, rely, make, are, seen, enabling, make                                  1
create, using, sharing, reflecting                                                  1
trained, demonstrated, achieve, flops, achieved, matching, learn                    1
look, is, chatgpt                                                                   1
Name: count, Length: 772, dtype: int64

In [13]:
dfTemp["adjectives"].value_counts()

adjectives
own, assistant, openai, api, cool             35
username                                      30
new                                           15
gpt3                                          15
gptjt                                         12
                                              ..
username, id, generic, other, large, i, ai     1
username, perfect, next, ridiculous            1
much, faster, least                            1
username, fantastic, openpit                   1
incredible, more, openai, ai                   1
Name: count, Length: 672, dtype: int64

In [14]:
dfTemp["subjects"].value_counts() #INTERESTING (when coding (or other words related to it) in the value)

subjects
i, assistant w gpt3, the plan, this, you                                 35
i                                                                        34
openai upgrades gpt3                                                     25
artist                                                                   19
chatgpt                                                                  13
                                                                         ..
openai upgrades, openai upgrades gpt3 with impressive new skills          1
gpt3 musk, this article, elon, musk, the article                          1
 applications artificialintelligence openai upgrades                      1
key insights from username interview with username peter welinder, we     1
chatgpt, this thing, education, chatgpt openai                            1
Name: count, Length: 635, dtype: int64

In [15]:
dfTemp["objects"].value_counts() #INTERESTING (when coding (or other words related to it) in the value)

objects
my own, openai playground connect, api ship, a web app chrome extension, it                                                                                                                                                                                                                                                    35
rhyming poetry and lyrics                                                                                                                                                                                                                                                                                                      24
childhood diary, gpt3, a moving chat with herself, herself                                                                                                                                                                                                                                                                     10
rhyming poetry andlyrics  

In [16]:
dfTemp["predicates"].value_counts()

predicates
username    120
is          107
stunning     42
working      38
feeds        22
           ... 
upgrade       1
seem          1
shares        1
npews         1
place         1
Name: count, Length: 389, dtype: int64

In [17]:
dfTemp["topic"].value_counts() #INTERESTING (when coding (or other words related to it) in the value)

topic
chatgpt, ai, will                                                                                                                                                                                            688
davinci, davinci003, textdavinci003, model, davinci model, gpt35turbo, api, textdavinci002, gpt3 davinci, cheaper, davinci3, chatgpt api, models, 10x cheaper, gpt3 textdavinci003, da vinci                  28
gpt4, ai, gpt3, deeplearning gpt, ai gpt, will, future, gpt ai, ai tech                                                                                                                                       21
openai, gpt4, openai gpt, api, openai has, gpt3, openai gpt4, trademark, gpt4 openai, model, gpt5, openai api, from openai, if openai, access, models, openai gpt3                                            18
parameters, model, training, models, gpt3, trillion, trained, gpt4, trillion parameters, gpt models, data, size, gpt5, training data, billion, billion paramet

### interesting columns list

In [18]:
goodColumns = ["technologies", "job_competencies", "job_profiles"]
goodColumnsWhenCodeRelated = [
                                #"objects", 
                                #"subjects", 
                                "topic"
                              ]

we prompted gpt with this question

"i have to filter a dataset of tweets by the ones related to coding, give me a list of 20 words that could be in tweets related to coding and generative ai"

and the following is the list returned which we will use to filter the dataset on the columns (+ some of ours)

In [19]:
words_related_to_coding_and_ai = [
    "Python",
    "JavaScript",
    "C",
    "cpp",
    "c++",
    "C#",
    "Ruby",
    "Swift",
    "Kotlin",
    "R",
    "Go",
    "Scala",
    "PHP",
    "HTML",
    "CSS",
    "SQL",
    "TypeScript",
    "Perl",
    "Assembly",
    "Objective-C",
    "Programming",
    "Code",
    #"AI",
    #"Machine Learning",
    #"Generative",
    #"Deep Learning",
    #"Neural Networks",
    "Algorithm",
    #"Data Science",
    #"TensorFlow",
    #"PyTorch",
    "GitHub",
    "Framework",
    #"Model",
    #"Development",
    #"Automation",
    #"Creative",
    #"Innovation",
    "Backend",
    "Frontend"
]

now we filter the dataset by where the good columns are not null

In [20]:
df = df[df[goodColumns].notnull().all(axis=1)]
df.shape

(74777, 17)

In [21]:
df = df.drop(columns=['Unnamed: 0', "original_text"])
df.to_csv("half filtered.csv", index=False)

we still have a big dataset so we can keep the filter and add the next one too

In [22]:
tempDf = df
for column in goodColumnsWhenCodeRelated:
    tempDf = tempDf[tempDf[column].str.contains('|'.join(words_related_to_coding_and_ai).lower(), na=False)]

tempDf.shape

error: multiple repeat at position 26

we can now save the df to file (we went from 3821843 records to 3806)

In [ ]:
tempDf.to_csv("final.csv", index=False)